dbt/adapters/athena/connections.py

```diff
@@ -35,6 +38,63 @@
 logger = AdapterLogger("Athena")
 
 
+def paginate_insert_into(func):
+    def wrapper(*args, **kwargs):
+        # get the args
+        sig = inspect.signature(func)
+        argmap = sig.bind_partial(*args, **kwargs).arguments
+        operation = argmap.get("operation")
+        # if it's insert into, paginate, else return
+        if operation.strip().upper().startswith("INSERT INTO"):
```

Comment:
Is it not a bit extreme to do this check for every query executed?

I understand the need to paginate but I feel it could have a cleaner implementation. Maybe support more than only `INSERT INTO`.

Reply:
As far as I can tell insert into is only called when the seed tables are being populated/updated which is where the problem comes from (lengthy string columns + bigger seed tables than perhaps than originally thought of) 
The implementation could for sure be cleaner, but we need to decide whether to paginate the query or not, currently the only reason I'd see to do it is if it's an insert into query, unless you have other ideas?

Reply:
@s-ducks check this here https://github.com/dbt-athena/dbt-athena/blob/8246464a729a375f59abc2ff931b39e038242a4d/dbt/include/athena/macros/materializations/models/incremental/helpers.sql#L31 it's called when doing incremental materialization.